core/io/file_access.cpp

```diff
@@ -71,7 +71,7 @@ void FileAccess::_set_access_type(AccessType p_access) {
 
 Ref<FileAccess> FileAccess::create_for_path(const String &p_path) {
 	Ref<FileAccess> ret;
-	if (p_path.begins_with("res://")) {
+	if (p_path.begins_with("res://") || p_path.begins_with("uid://")) {
```

Comment:
I really think file layer should have nothing to do with UIDs, the extentof UIDs should be only resources.